### PR TITLE
Move 3rd party propagators and merge exporter into sdk::export

### LINKED ISSUE #340: Consider moving `exporters` module to sdk
It may be better to have the exporters in the sdk to clarify that they are not part of the API. Could move to https://github.com/open-telemetry/opentelemetry-rust/tree/master/opentelemetry/src/sdk/export

### LINKED ISSUE #361: Move 3rd party propagators to contrib
See the spec change for details https://github.com/open-telemetry/opentelemetry-specification/pull/1144

### PR DESCRIPTION
**Pull request by TommyCpp**
Fix #361 
Fix #340

This PR:
 - Moved 3rd party propagators into `opentelemetry-contrib/propagator`
 - Added `testing` feature so that we could share test util between crates.
 - Moved `exporter` into `sdk::export`
 - Updated the `opentelemetry-proto`

Upgrade note:
 - For `aws` propagators. Add `opentelemetry-contrib` in your `Cargo.toml`. Choose the feature based on the propagators you need and import them via `opentelemetry_contrib::trace::propagators::<propagators name>`.
 - For types from `exporter` module. Such as `ExportError`. Change the import path from `opentelemetry::exporter` into `opentelemetry::sdk::export`
- For `jaeger` propagators. Add `opentelemetry-jaeger` in your `Cargo.toml`. use `opentelemetry_jaeger::Propagator`
- For `b3` propagators. Add `opentelemetry-zipkin` in your `Cargo.toml`. use `opentelemetry_zipkin::Propagator`

Open question(s):
 - <del>Where should we put the `jaeger` and `b3` propagtor? It's intutive to me that we include them as part of the `opentelemetry-jaeger` or `opentelemetry-zipkin` crate. But Go impl put it in `contrib`</del> Moved zipkin and jaeger propagators into their own crate.

### CODE REVIEW

**Comment by jtescher** [comment]
Hm yeah I think it may be more intuitive to have the jaeger / be propagators with the exporters in their crates rather than contrib.

**Review by jtescher — APPROVED** [review]
Other than that I think this looks good

File: opentelemetry-contrib/src/trace/propagator/mod.rs [hunk 1]
```diff
@@ -6,4 +6,14 @@
 //! Currently, the following propagators are supported:
 //!
 //! * `binary_propagator`, propagating trace context in the binary format.
+//! * `XrayPropagator`, propagating via AWS XRay protocol.
+//! * `B3Propagator`, propagating via B3 protocol and headers.
```
Comment (jtescher) [inline opentelemetry-contrib/src/trace/propagator/mod.rs:10]: These docs can be moved now I think too

Comment (jtescher) [reply opentelemetry-contrib/src/trace/propagator/mod.rs:10]: (the non-xray ones)

Comment (TommyCpp) [reply opentelemetry-contrib/src/trace/propagator/mod.rs]: fixed

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -129,3 +129,7 @@ See the [code owners](CODEOWNERS) file.
 
 See the [community membership document in OpenTelemetry community
 repo](https://github.com/open-telemetry/community/blob/master/community-membership.md).
+
+## FAQ
+### Where should I put third party propagators/exporters, contrib or standalone crates?
```
Comment (TommyCpp) [inline CONTRIBUTING.md:134]: wanted to clarify the problem that where to put the stuff here. Please let me know if the below descriptions makes sense. @jtescher

Comment (jtescher) [reply CONTRIBUTING.md:134]: Seems good 👍

**Pull request merged**